### DISCUSSION TITLE
[BUG] Immich fails to start when the external library is offline

### DISCUSSION THREAD

**Discussion by travelr** [immich-app/immich#4805] [Q&A] [answered]
### The bug

Hi,

when a external library (mount) is not available then Immich fails to start via docker. I run into this issue mainly when Watchtower is updating Immich. Then it fails to start until you restore the connection to the library. 

Is it possible to make external libraries fail silent if the mount is not available?

```
Found new [ghcr.io/immich-app/immich-proxy:release](http://ghcr.io/immich-app/immich-proxy:release) image (7a8db6b05f20)
Found new [ghcr.io/immich-app/immich-web:release](http://ghcr.io/immich-app/immich-web:release) image (7582947032dd)
Found new [ghcr.io/immich-app/immich-server:release](http://ghcr.io/immich-app/immich-server:release) image (00a6abcca4cd)
Found new [ghcr.io/immich-app/immich-server:release](http://ghcr.io/immich-app/immich-server:release) image (00a6abcca4cd)
Stopping /immich_server (29882f88facf) with SIGTERM
Stopping /immich_microservices (15ed80080880) with SIGTERM
Stopping /immich_web (ffac222af448) with SIGTERM
Stopping /immich_proxy (d8a08f4cb2a7) with SIGTERM
Creating /immich_proxy
Creating /immich_web
Error response from daemon: failed to create task for container: failed to create shim task: OCI runtime create failed: runc create failed: unable to start container process: error during container init: error mounting "/mnt/photos_ext" to rootfs at "/mnt/media/photos": mount /mnt/photos_ext:/mnt/media/photos (via /proc/self/fd/6), flags: 0x5001: no such device: unknown
Creating /immich_microservices
Error response from daemon: failed to create task for container: failed to create shim task: OCI runtime create failed: runc create failed: unable to start container process: error during container init: error mounting "/mnt/photos_ext" to rootfs at "/mnt/media/photos": mount /mnt/photos_ext:/mnt/media/photos (via /proc/self/fd/6), flags: 0x5001: no such device: unknown
Creating /immich_server
Error response from daemon: failed to create task for container: failed to create shim task: OCI runtime create failed: runc create failed: unable to start container process: error during container init: error mounting "/mnt/photos_ext" to rootfs at "/mnt/media/photos": mount /mnt/photos_ext:/mnt/media/photos (via /proc/self/fd/6), flags: 0x5001: no such device: unknown
```

PS: If anybody can tell me why the CIFS mount to Hetzner stops working after a day I would appreciate it.
```
//123.your-storagebox.de/123-sub1 /mnt/photos_ext_crypt cifs credentials=/root/.smb_cred_photos,iocharset=utf8,rw,_netdev,uid=xy,gid=xy,file_mode=0660,dir_mode=0770,vers=3.0  0 0
```

### The OS that Immich Server is running on

Ubuntu 22

### Version of Immich Server

latest

### Version of Immich Mobile App

latest

### Platform with the issue

- [X] Server
- [X] Web
- [ ] Mobile

### Your docker-compose.yml content

```YAML
volumes:
    - /home/xy/photos:/usr/src/app/upload
    - /mnt/photos_ext:/mnt/media/photos:ro
```

### Your .env content

```Shell
-
```

### Reproduction steps

```bash
1.
2.
3.
...
```

**Comment by alextran1502** [comment]
I would treat this as a dependency to use external library feature, you would need the library to be mounted correctly.

For CIFS mount, this is a bug in the latest NodeJS releases (context here https://github.com/nodejs/node/issues/50061)

The work around is to add `UV_USE_IO_URING=0` to the .env file

**Comment by bo0tzz** [comment]
Converted to a discussion as this is not strictly a bug with Immich, but rather a detail of how docker handles mounts. If the external library is empty or not mounted to the container, Immich will still start up successfully. I don't know why exactly you're getting the error that you are, but my guess would be that `/mnt/photos_ext` does not exist at all until it's mounted on your host system. 

As a sidenote, it's not recommended to use watchtower for updating Immich as there are often breaking changes that require manual action.

**Comment by travelr** [comment]
Thanks guys for the immediate response.  I will look into the env variable.

The mount does exit, but the response is "Host down"

I agree this is not an immich issue directly. Nonetheless my intent of the ticket was if a start script of the container could handle this and fail silently?

**Reply by bo0tzz** [reply, marked as answer]
A start script wouldn't help here, as Docker isn't even starting the container to begin with.
